app/helpers/dashboard_helper.rb

```diff
@@ -3,15 +3,25 @@ module DashboardHelper
   def dashboard_actions
     [_("Generated at %s") % Time.zone.now.to_s(:short),
      select_action_button(_("Manage dashboard"), {},
-                          link_to_function(_("Save dashboard"), 'save_position()'),
-                          link_to_function(_("Reset to default"), 'reset_position()'),
+                          link_to_function(_("Save dashboard"), "save_position('#{save_positions_widgets_path}')"),
+                          link_to(_("Reset to default"), reset_default_widgets_path),
                           content_tag(:li,'',:class=>'divider'),
                           content_tag(:li,_("Restore widgets"), :class=>'nav-header', :id=>'restore_list' )
      )]
   end
 
-  def widget_list
-    Dashboard::Manager.widgets
+  def render_widget(widget)
+    if widget.data.include?('host_id')
```

Comment:
Something along the lines of:

```ruby
widget.data.each do |datum|
  next unless /_id/ =~ datum
  instance_variable_set(:"@#{datum[0..-4]}", datum[0..-4].classify.constantize.find(datum))
end
```

would allow to add future _id widgets easily